app/lib/frontend/templates/package_versions.dart

```diff
@@ -24,23 +24,27 @@ String renderPkgVersionsPage(
 
   final stableVersionRows = [];
   final devVersionRows = [];
-  PackageVersion latestDevVersion;
+  final latestDevVersion = versions.firstWhere(
+    (v) => v.version == data.package.latestDevVersion,
+    orElse: () => null,
+  );
   for (int i = 0; i < versions.length; i++) {
     final PackageVersion version = versions[i];
     final String url = versionDownloadUrls[i].toString();
     final rowHtml = renderVersionTableRow(version, url);
     if (version.semanticVersion.isPreRelease) {
-      latestDevVersion ??= version;
       devVersionRows.add(rowHtml);
     } else {
       stableVersionRows.add(rowHtml);
     }
   }
 
   final htmlBlocks = <String>[];
-  if (stableVersionRows.isNotEmpty && devVersionRows.isNotEmpty) {
+  if (stableVersionRows.isNotEmpty &&
+      devVersionRows.isNotEmpty &&
+      data.package.showDevVersion) {
     htmlBlocks.add(
-        '<p>The latest dev release was <a href="#dev">${latestDevVersion.version}</a> '
+        '<p>The latest pre-release was <a href="#dev">${latestDevVersion.version}</a> '
```

Comment:
@kwalrath Do you have a position on "prereleases" vs. "pre-releases". I recall you pushing us to use `--prereleases` instead of `--pre-releases` for `pub outdated`. And your docs use "prereleases" everywhere.

But [semantic versioning](https://semver.org/) where the term comes from uses "pre-releases". I'm happy either way, but it's probably best we're internally consistent in Dart :)

Reply:
Yep, we use _prerelease_, no hyphen.

Here's how I decide on things like this. If it isn't in the [Google dev word list](https://developers.google.com/style/wordlist), I follow the advice of the [Google dev spelling guide](https://developers.google.com/style/spelling) and look it up on https://www.merriam-webster.com. https://www.merriam-webster.com/dictionary/prerelease lists the no-hyphen term first, so I'm going with that. Also, Google has a [preference not to hyphenate](https://developers.google.com/style/hyphens#when-to-hyphenate).